File: Algorithms/p376_Wiggle_Subsequence_1.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) {
        if(nums.empty()) return 0;
        int up = 1; // max length of sequences with increasing tail so far
        int down = 1; // max length of sequences with decreasing tail so far
        for(int i=1;i<nums.size();++i){
            if(nums[i]>nums[i-1]){
                up = down+1;
            }else if(nums[i]<nums[i-1]){
                down = up+1;
            }
        }
        return max(up,down);
    }
};

File: Algorithms/p376_Wiggle_Subsequence/p376_Wiggle_Subsequence_1.py
class Solution(object):
    def wiggleMaxLength(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        up = 1 # max length of sequences with increasing tail so far
        down = 1 # max length of sequences with decreasing tail so far
        for i in range(1,len(nums)):
            if nums[i]>nums[i-1]:
                up = down+1
            elif nums[i]<nums[i-1]:
                down = up+1
        return max(up,down)
